internal/api/personal_access_tokens/2024-03-19/pat.go

```diff
@@ -0,0 +1,187 @@
+package v20240319
+
+import (
+	"encoding/json"
+	"fmt"
+	"time"
+
+	"github.com/oklog/ulid"
+)
+
+// Enum of PAT type.
+const (
+	TypePAT Type = iota + 1
+)
+
+var (
+	typeToString = map[Type]string{
+		TypePAT: "personal_access_token",
+	}
+
+	stringToType = map[string]Type{
+		"personal_access_token": TypePAT,
+	}
+)
+
+// Type represents the different types of PAT.
+type Type int
+
+// ParseType parses a string representing a type and returns a Type if valid.
+// It returns an error if the string representation is not a valid type.
+func ParseType(str string) (t Type, err error) {
```

Comment:
are the named return parameters wanted? They can easily lead to unexpected behaviour. See https://www.reddit.com/r/golang/comments/1ajuhlr/weird_behavior_with_named_return_variables/

Reply:
[updated](https://github.com/snyk/go-application-framework/pull/348#discussion_r2100701018) to generate the models via the spec